xla/service/gpu/gpu_fusible.cc

```diff
@@ -878,16 +878,16 @@ size_t GetOutputSizeOfFusible(const HloInstruction& instr) {
 // Recursive helper for GetFusionRoots below.
 static void GetFusionRootsRec(const HloInstruction* root,
                               std::vector<const HloInstruction*>& out) {
-  if (root->opcode() == HloOpcode::kGetTupleElement) {
+  if (root->opcode() == HloOpcode::kGetTupleElement &&
+      root->operand(0)->opcode() == HloOpcode::kTuple) {
```

Comment:
Your test case GpuFusibleTest.GetFusionRootsWithMakeTupleGTESequence shows that we will need an additional change to make this work correctly. We should only access the part of the tuple that the get-tuple-element refers to. So:

```
return GetFusionRootsRec(root->operand(0)->operand(root->tuple_index()), out);
```